src/nvim/normal.c

```diff
@@ -5034,6 +5035,15 @@ static void nv_ident(cmdarg_T *cap)
     g_tag_at_cursor = true;
     do_cmdline_cmd(buf);
     g_tag_at_cursor = false;
+
+    if (cmdchar == 'K' && !kp_ex && !kp_help) {
+      // Start insert mode in terminal buffer
+      restart_edit = 'i';
+
+      add_map((char_u *)"<buffer> <esc> <Cmd>call jobstop(&channel)<CR>", TERM_FOCUS, true);
+      do_autocmd_event(EVENT_TERMCLOSE, (char_u *)"<buffer>", false, false, (char_u *)"bdelete!",
+                       false, AUGROUP_DEFAULT);
```

Comment:
can we just use `do_cmdline_cmd` here? That avoids exposing do_autocmd_event(). Performance of `do_cmdline_cmd` is small compared to the cost of the rest of this codepath.

Reply:
Sure, I'll just use `do_autocmd` (it parses a string like `do_cmdline_cmd` so is equally readable).